idaes/power_generation/unit_models/steamheater.py

```diff
@@ -0,0 +1,720 @@
+                         doc="Angle at joint of tube "
+                         "and fin at outside slag layer")
+        def alpha_slag(b, t):
+            return asin((b.thk_fin_half + b.slag_thickness[t]) /
+                        (b.radius_out + b.slag_thickness[t]))
+
+        @self.Expression(doc="Perimeter of interface between slag and tube")
+        def perimeter_if(b):
+            if self.config.single_side_only:
+                return (const.pi - 2 * b.alpha_tube) * b.radius_out + b.pitch \
+                    - 2 * b.radius_out * cos(b.alpha_tube)
+            else:
+                return 2*((const.pi-2*b.alpha_tube) * b.radius_out + b.pitch
+                          - 2 * b.radius_out * cos(b.alpha_tube))
+
+        @self.Expression(doc="Perimeter on the inner tube side")
+        def perimeter_ts(b):
+            return const.pi*b.diameter_in
+
+        @self.Expression(self.flowsheet().config.time,
+                         doc="Perimeter on the outer slag side")
+        def perimeter_ss(b, t):
+            if self.config.single_side_only:
+                return (const.pi - 2 * b.alpha_slag[t]) \
+                    * (b.radius_out + b.slag_thickness[t]) + \
+                    b.pitch - 2 * (b.radius_out + b.slag_thickness[t]) \
+                    * cos(b.alpha_slag[t])
+            else:
+                return 2 * ((const.pi - 2 * b.alpha_slag[t])
+                            * (b.radius_out + b.slag_thickness[t])
+                            + b.pitch - 2 * (b.radius_out
+                                             + b.slag_thickness[t])
+                            * cos(b.alpha_slag[t]))
+
+        # Cross section area of tube and fin metal
+        @self.Expression(doc="Cross section area of tube and fin metal")
+        def area_cross_metal(b):
+            return const.pi * (b.radius_out**2 - b.radius_in**2) \
+                + b.fin_thickness * b.fin_length
+
+        # Cross section area of slag layer
+        @self.Expression(self.flowsheet().config.time,
+                         doc="Cross section area of slag layer per tube")
+        def area_cross_slag(b, t):
+            return b.perimeter_if * b.slag_thickness[t]
+
+        # Volume constraint
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="waterwall fluid volume of all tubes")
+        def volume_eqn(b, t):
+            return b.volume[t] == 0.25 * const.pi * b.diameter_in**2 \
+                * b.tube_length * b.number_tubes
+
+    def _make_performance(self):
+        """
+        Define constraints which describe the behaviour of the unit model.
+        """
+        # Thermal conductivity of metal
+        self.therm_cond_metal = Param(
+                initialize=43.0,
+                mutable=True,
+                doc='Thermal conductivity of tube metal')
+        # Thermal conductivity of slag
+        self.therm_cond_slag = Param(
+                initialize=1.3,
```

Comment:
For the future, but these would probably be better as `Vars` to allow for parameter estimation (slag conductivity strikes me as something we would want to estimate).